doc-content/shared-kie-docs/src/main/asciidoc/KieServer/controller-java-api-con.adoc

```diff
@@ -0,0 +1,191 @@
+[id='controller-java-api-con_{context}']
+= {CONTROLLER} Java client API for {KIE_SERVER} templates and instances
+
+{PRODUCT} provides a {CONTROLLER} Java client API that you can use to send REST or WebSocket requests to the {CONTROLLER} from your Java client application. You can use the {CONTROLLER} Java client API as an alternative to the {CONTROLLER} REST API to interact with your {KIE_SERVER} templates (configurations), {KIE_SERVER} instances (remote servers), and associated KIE containers (deployment units) in {PRODUCT} without using {CENTRAL}. This API support enables you to maintain your {PRODUCT} resources more efficiently and optimize your integration and development with {PRODUCT}.
+
+With the {CONTROLLER} Java client API, you can perform the following actions also supported by the {CONTROLLER} REST API:
+
+* Retrieve information about {KIE_SERVER} templates, instances, and associated KIE containers
+* Update, start, or stop KIE containers associated with {KIE_SERVER} templates and instances
+* Create, update, or delete {KIE_SERVER} templates
+* Create, update, or delete {KIE_SERVER} instances
+
+{CONTROLLER} Java client API requests require the following components:
+
+Authentication::
+The {CONTROLLER} Java client API requires basic HTTP Authentication for the following user roles, depending on controller type:
++
+--
+* `rest-all` user role if you installed {CENTRAL} and you want to use the built-in {CONTROLLER}
+* `kie-server` user role if you installed the {HEADLESS_CONTROLLER} separately from {CENTRAL}
+
+To view configured user roles for your {PRODUCT} distribution, navigate to `~/$SERVER_HOME/standalone/configuration/application-roles.properties` and `../application-users.properties`.
+
+To add a user with the `kie-server` role or the `rest-all` role or both, navigate to `~/$SERVER_HOME/bin` and run the following command with the role or roles specified:
+
+[source,bash]
+----
+$ ./add-user.sh -a --user <USERNAME> --password <PASSWORD> --role kie-server,rest-all
+----
+
+To configure the `kie-server` or `rest-all` user with {CONTROLLER} access, navigate to `~/$SERVER_HOME/standalone/configuration/standalone-full.xml`, uncomment the `org.kie.server` properties (if applicable), and add the controller user login credentials and controller location (if needed):
+
+[source,xml,subs="attributes+"]
```

Comment:
You dont really need to do this xml change in order to simply connect from the client to the controller. As long as the user has the credentials mentioned above it should work.

Reply:
Understood. I'm just trying to be consistent with our documentation elsewhere, which does walk through this. Trying to set the reader/user up for as much success as possible.